ui/imports/shared/status/StatusChatInput.qml

```diff
@@ -688,45 +709,17 @@ Rectangle {
             d.leftOfMentionIndex = -1
         }
 
-        messageInputField.readOnly = false
-
-        if (d.internalPaste) {
-            if (d.copiedTextPlain.includes("@")) {
-                d.copiedTextFormatted = d.copiedTextFormatted.replace(/span style="/g, "span style=\" text-decoration:none;")
-
-                let lastFoundIndex = -1
-                for (let j = 0; j < d.copiedMentionsPos.length; j++) {
-                    const name = d.copiedMentionsPos[j].name
-                    const indexOfName = d.copiedTextPlain.indexOf(name, lastFoundIndex)
-                    lastFoundIndex += name.length
-
-                    if (indexOfName === d.copiedMentionsPos[j].leftIndex + 1) {
-                        const mention = {
-                            name: name,
-                            pubKey: d.copiedMentionsPos[j].pubKey,
-                            leftIndex: (d.copiedMentionsPos[j].leftIndex + d.copyTextStart - 1),
-                            rightIndex: (d.copiedMentionsPos[j].leftIndex + d.copyTextStart + name.length)
-                        }
-                        mentionsPos.push(mention)
-                        d.sortMentions()
-                    }
-                }
-            }
-
-            const prevLength = messageInputField.length
-            insertInTextInput(d.copyTextStart, d.copiedTextFormatted)
-            messageInputField.cursorPosition = d.copyTextStart + messageInputField.length - prevLength
-            d.internalPaste = false
-        } else if (event.matches(StandardKey.Paste) && QClipboardProxy.hasText) {
-            messageInputField.insert(d.copyTextStart, "<div style='white-space: pre-wrap'>" + Utils.escapeHtml(QClipboardProxy.text) + "</div>") // preserve formatting
-        }
-
         if (event.key !== Qt.Key_Escape) {
             emojiEvent = emojiHandler(event)
             if (!emojiEvent) {
                 emojiSuggestions.close()
             }
         }
+
+        if (messageInputField.readOnly) {
+            messageInputField.readOnly = false;
+            messageInputField.cursorPosition = d.copyTextStart + QClipboardProxy.text.length;
```

Comment:
How do you know at this point that you pasted a text and not an image?

Reply:
`messageInputField.readOnly` will be only set to true in line 456, so inside `else if (QClipboardProxy.hasText) {`, I guess this should be enough?

Reply:
Alright... eventhough setting it to `readOnly` itself is cringy :)